src/NavBar.ts

```diff
@@ -112,60 +88,78 @@ export default class NavBar {
         return ''
     }
 
-    private parseLayer(url: URL) {
-        let layer = url.searchParams.get('layer')
-        const option = this.mapStore.state.styleOptions.find(option => option.name === layer)
-        return option ? option : this.mapStore.state.selectedStyle
+    private static parseLayer(url: URL): string | null {
+        return url.searchParams.get('layer')
     }
 
-    parseUrlAndReplaceQuery() {
-        this.isIgnoreQueryStoreUpdates = true
+    async updateStateFromUrl() {
+        // We update the state several times ourselves, but we don't want to push history entries for each dispatch.
+        this.ignoreStateUpdates = true
 
         Dispatcher.dispatch(new ClearPoints())
         const url = new URL(window.location.href)
 
         const parsedProfileName = NavBar.parseProfile(url)
-        const profile = parsedProfileName ? { name: parsedProfileName } : this.queryStore.state.routingProfile
-        const parsedPoints = this.parsePoints(url, profile)
-
-        // estimate map bounds from url points if there are any. this way we prevent loading tiles for the world view
-        // only to zoom to the route shortly after
-        const bbox = this.getBBoxFromUrlPoints(parsedPoints.map(p => p.coordinate))
-        if (bbox) Dispatcher.dispatch(new SetInitialBBox(bbox))
+        if (parsedProfileName)
+            // this won't trigger a route request because we just cleared the points
+            Dispatcher.dispatch(new SetVehicleProfile({ name: parsedProfileName }))
+        const parsedPoints = NavBar.parsePoints(url)
 
-        if (parsedPoints.length > 0) Dispatcher.dispatch(new SetRoutingParametersAtOnce(parsedPoints, profile))
+        // support legacy URLs without coordinates (not initialized) and only text, see #199
+        if (parsedPoints.some(p => !p.isInitialized && p.queryText.length > 0)) {
+            const promises = parsedPoints.map(p => {
+                if (p.isInitialized) return Promise.resolve(p)
+                return (
+                    getApi()
+                        .geocode(p.queryText, 'nominatim')
+                        .then(res => {
+                            if (res.hits.length == 0) return p
+                            return {
+                                ...p,
+                                queryText: res.hits[0].name,
+                                coordinate: { lat: res.hits[0].point.lat, lng: res.hits[0].point.lng },
+                                isInitialized: true,
+                            }
+                        })
+                        // if the geocoding request fails we just keep the point as it is, just as if no results were found
+                        .catch(() => Promise.resolve(p))
```

Comment:
We can also do something like http://localhost:3000/?point=&point=Berlin now which will resolve the second point from the 'Berlin' query string and the first will just be an empty input box.